lib/xo.ts

```diff
@@ -346,8 +373,9 @@ export class Xo {
 			throw new Error('Failed to initialize ESLint');
 		}
 
```

Comment:
Maybe we can simplify this PR. 

Add the `combinedIgnores` to the original globby lookup to get `files`. The `combinedIgnores` needs to check whether the ignore is the only key: see https://github.com/xojs/xo/blob/main/lib/xo-to-eslint.ts#L34

Move the `checkForIgnoredDirectFiles` after the files look up and make sure it all is does is use fs.stat to filter out the direct files, then check if each direct file is included in `files` and if not, then we can throw our error.